### PR TITLE
Add log entries for proxy configuration

### PR DESCRIPTION
**Pull request by juzuluag**
Issue Number: 323

## Summary
Add log entries with regards proxy configuration
Fix web proxy instantiation
Clean up double call to service collection registration

## Changes

- Add log for proxy configuration

## Checklist

- [x] Local Tests Passing?
- [x] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [x] This is not a breaking change. If it is, please describe it below.

## Is this a breaking change?
N/A

### CODE REVIEW

**Review by rakker91 — APPROVED** [review]

File: src/dotnet/CarbonAware.Tools.WattTimeClient/Configuration/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -50,4 +55,14 @@ public static IServiceCollection ConfigureWattTimeClient(this IServiceCollection
...
         return services;
     }
+
+    private static void LogProxyConfiguration(IConfiguration config, CarbonAwareVariablesConfiguration caVars)
+    {
+        ILoggerFactory factory = LoggerFactory.Create(b => {
+            b.AddConfiguration(config.GetSection("Logging"));
+            b.AddConsole();
+        });
+        var logger = factory.CreateLogger<IServiceCollection>();
```
Comment (pritipath) [inline src/dotnet/CarbonAware.Tools.WattTimeClient/Configuration/ServiceCollectionExtensions.cs:65]: I am not sure if we should log the username. If we really wish to log it for debugging purposes, maybe we log a hash of username.

Comment (juzuluag) [reply src/dotnet/CarbonAware.Tools.WattTimeClient/Configuration/ServiceCollectionExtensions.cs:65]: No preference here. I think we are only doing this for troubleshooting purposes, so hashing the username would require that the consumer/client knows what the hash is about.

Comment (pritipath) [reply src/dotnet/CarbonAware.Tools.WattTimeClient/Configuration/ServiceCollectionExtensions.cs:65]: Makes sense. I still dont think it is advisable to log username, but I am fine with whatever the team thinks is right. Will resolve the conversation.

**Review by JenMadiedo — APPROVED** [review]
🚢 -it

**Review by gfmatthews — APPROVED** [review]

**Pull request merged**